brush-shell/tests/cases/subshell.yaml

```diff
@@ -0,0 +1,33 @@
+name: "subshell"
```

Comment:
_Minor comment:_ it's true that `$(...)` gets run in a subshell, but we've found it useful to specifically refer to it as a "command substitution" (following the [POSIX docs](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03)) to distinguish it from other constructs that also end up executing logic in subshells (e.g., `(...)`).

What do you think about renaming the file to `command_substitution.yaml`? We already have some command substitution tests in `word_expansion.yaml` but that file has gotten large and unwieldy. I like the idea of factoring out smaller `.yaml` files and, over time, breaking it up. (Doesn't need to be done in this PR.)